apps/challenges/aws_utils.py

```diff
@@ -1109,6 +1110,251 @@ def create_eks_nodegroup(challenge, cluster_name):
+        )
+        ecr_all_access_policy_arn = response["Policy"]["Arn"]
+        waiter = client.get_waiter("policy_exists")
+        waiter.wait(PolicyArn=ecr_all_access_policy_arn)
+        # Attach custom ECR policy
+        response = client.attach_role_poilcy(
+            RoleName=node_group_role_name, PolicyArn=ecr_all_access_policy_arn
+        )
+    except ClientError as e:
+        logger.exception(e)
+        return response
+    try:
+        challenge_evaluation_cluster = ChallengeEvaluationCluster.objects.get(
+            challenge_obj
+        )
+        serializer = ChallengeEvaluationClusterSerializer(
+            challenge_evaluation_cluster,
+            data={
+                "eks_arn_role": eks_arn_role,
+                "node_group_arn_role": node_group_arn_role,
+                "ecr_all_access_policy_arn": ecr_all_access_policy_arn,
+            },
+            partial=True,
+        )
+        if serializer.is_valid():
+            serializer.save()
+        # Create eks cluster vpc and subnets
+        create_eks_cluster_subnets.delay(challenge)
+    except Exception as e:
+        logger.exception(e)
+        return
+
+
+@app.task
+def create_eks_cluster_subnets(challenge):
+    """
+    Creates EKS and NodeGroup ARN roles
+
+    Arguments:
+        instance {<class 'django.db.models.query.QuerySet'>} -- instance of the model calling the post hook
+    """
+    from .models import ChallengeEvaluationCluster
+    from .serializers import ChallengeEvaluationClusterSerializer
+    from .utils import get_aws_credentials_for_challenge
+
+    for obj in serializers.deserialize("json", challenge):
+        challenge_obj = obj.object
+    challenge_aws_keys = get_aws_credentials_for_challenge(challenge_obj.pk)
+    client = get_boto3_client("ec2", challenge_aws_keys)
+    vpc_ids = []
+    try:
+        response = client.create_vpc(CidrBlock="100.68.0.0/16")
+        vpc_ids.append(response["Vpc"]["VpcId"])
+    except ClientError as e:
+        logger.exception(e)
+        return response
+
+    waiter = client.get_waiter("vpc_available")
+    waiter.wait(VpcIds=vpc_ids)
+
+    # Create internet gateway and attach to vpc
+    try:
+        response = client.create_internet_gateway()
+        internet_gateway_id = response["InternetGateway"]["InternetGatewayId"]
+        client.attach_internet_gateway(
+            InternetGatewayId=internet_gateway_id, VpcId=vpc_ids[0]
+        )
+
+        # Create and attach route table
+        response = client.create_route_table(VpcId=vpc_ids[0])
+        route_table_id = response["RouteTable"]["RouteTableId"]
+        client.create_route(
+            DestinationCidrBlock="0.0.0.0/0",
+            GatewayId=internet_gateway_id,
+            RouteTableId=route_table_id,
+        )
+
+        # Create subnets
+        subnet_ids = []
+        response = client.create_subnet(
+            CidrBlock="100.68.0.0/20",
+            AvailabilityZone="us-east-1a",
```

Comment:
What if `us-east-1a` doesn't support it? Is it not possible to pass a list of availability zones?

Reply:
We can only pass 1 availability zone. We could've allowed boto3 to auto assign availability zone but for EKS we need subnets in two availability zone. We can accept these from challenge config same as CIDRs

Reply:
okay, let's add a TODO above to replace this as well.

Reply:
Added it